tests/semantic_analyser.cpp

```diff
@@ -15,238 +16,251 @@ namespace bpftrace::test::semantic_analyser {
+  return test(
+      *bpftrace, true, input, expected_result, {}, safe_mode, has_child);
 }
 
-void test(std::string_view input, int expected_result)
+ast::ASTContext test(const std::string &input, int expected_result)
 {
   // This function will eventually be deprecated in favour of test_error()
   assert(expected_result != 0 &&
-         "Use test(std::string_view) for expected successes");
+         "Use test(const std::string&) for expected successes");
   auto bpftrace = get_mock_bpftrace();
-  Driver driver(*bpftrace);
-  test(*bpftrace, true, driver, input, expected_result, {}, true, false);
+  return test(*bpftrace, true, input, expected_result, {}, true, false);
 }
 
-void test(std::string_view input)
+ast::ASTContext test(const std::string &input)
 {
   auto bpftrace = get_mock_bpftrace();
-  Driver driver(*bpftrace);
-  test(*bpftrace, true, driver, input, 0, {}, true, false);
+  return test(*bpftrace, true, input, 0, {}, true, false);
 }
 
-void test(BPFtrace &bpftrace,
-          std::string_view input,
-          std::string_view expected_ast)
+ast::ASTContext test(BPFtrace &bpftrace,
+                     const std::string &input,
+                     std::string_view expected_ast)
 {
-  Driver driver(bpftrace);
-  test(bpftrace, true, driver, input, 0, {}, true, false);
+  auto ast = test(bpftrace, true, input, 0, {}, true, false);
 
   if (expected_ast[0] == '\n')
     expected_ast.remove_prefix(1); // Remove initial '\n'
 
   std::ostringstream out;
   ast::Printer printer(out);
-  printer.visit(driver.ctx.root);
+  printer.visit(ast.root);
 
   if (expected_ast[0] == '*' && expected_ast[expected_ast.size() - 1] == '*') {
     // Remove globs from beginning and end
     expected_ast.remove_prefix(1);
     expected_ast.remove_suffix(1);
     EXPECT_THAT(out.str(), HasSubstr(expected_ast));
-    return;
+    return ast;
   }
 
   EXPECT_EQ(expected_ast, out.str());
+  return ast;
 }
 
-void test(std::string_view input, std::string_view expected_ast)
+ast::ASTContext test(const std::string &input, std::string_view expected_ast)
 {
   auto bpftrace = get_mock_bpftrace();
-  test(*bpftrace, input, expected_ast);
+  return test(*bpftrace, input, expected_ast);
 }
 
-void test_error(BPFtrace &bpftrace,
-                std::string_view input,
-                std::string_view expected_error,
-                bool has_features = true)
+ast::ASTContext test_error(BPFtrace &bpftrace,
+                           const std::string &input,
+                           std::string_view expected_error,
+                           bool has_features = true)
 {
-  Driver driver(bpftrace);
-  test(bpftrace, has_features, driver, input, -1, expected_error, true, false);
+  return test(bpftrace, has_features, input, -1, expected_error, true, false);
 }
 
-void test_error(std::string_view input,
-                std::string_view expected_error,
-                bool has_features = true)
+ast::ASTContext test_error(const std::string &input,
```

Comment:
Just curious, what was the issue with string_view?

Reply:
Since the `ASTContext` requires a `const std::string&` input, I just made these consistent. If they aren't being modified, then there is no material difference between a `const std::string &` input and a `string_view`, the `string_view` can just be adjusted. (But this wasn't something we were doing anyways.)